SGoopas/Assets/Scripts/Game/EnemyCollisionHandler.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyCollisionHandler { 
+    public delegate void EnemyCollisionOccured(GameObject enemy);
+    public static event EnemyCollisionOccured EnemyCollisionEvent;
+    public static void TriggerEnemyCollision(GameObject enemy)
+    {
+        EnemyCollisionOccured enemyCol = EnemyCollisionEvent;
```

Comment:
pretty sure you can just call EnemyCollisionEvent(enemy), correct me if I'm wrong

Reply:
You can, but that's not thread-safe, while creating a temporary delegate copy of the event is. But unity isn't multithreaded, so I guess I could just call the event directly.

Reply:
well we don't really bother with that for other events like this, so I see no reason to start now